packages/devtools_app/lib/src/screens/memory/panes/diff/controller/diff_pane_controller.dart

```diff
@@ -61,6 +66,8 @@ class DiffPaneController {
   void deleteCurrentSnapshot() {
     assert(selected is SnapshotListItem);
     snapshots.removeRange(selectedIndex.value, selectedIndex.value + 1);
+    // We must change the selectedIndex, because otherwise the content will
+    // not be re-rendered.
     selectedIndex.value = selectedIndex.value - 1;
```

Comment:
this is not safe as is. what if 0 is the selected index? I know right now this is the "information" item, but if that changes and we choose to show instructions / info in a different way, we will have broken behavior. Use `math.max(selectedIndex.value - 1, 0)` instead.